### PR TITLE
28575588 update structured data response

### PR DESCRIPTION
**Pull request by iseabock**
### Motivation and Context

This PR adds frontend support for structured data responses.  
 
### Description

Screenshot 2024-08-07 133343

### Contribution Checklist

- [x] I have built and tested the code locally and in a deployed app
- [x] For frontend changes, I have pulled the latest code from main, built the frontend, and committed all static files.
- [x] This is a change for all users of this app. No code or asset is specific to my use case or my organization.
- [x] I didn't break any existing functionality :smile:

### CODE REVIEW

File: .github/workflows/28575588-update-structured-data-response_oyd-msi-structured-data-07032024.yml
Comment (sarah-widder) [inline .github/workflows/28575588-update-structured-data-response_oyd-msi-structured-data-07032024.yml:1]: I don't think these are needed, let's remove them

File: README.md [hunk 1]
```diff
@@ -265,7 +265,9 @@ Note: settings starting with `AZURE_SEARCH` are only needed when using Azure Ope
 |PROMPTFLOW_RESPONSE_FIELD_NAME|reply|Default field name to process the response from Promptflow request.|
 |PROMPTFLOW_CITATIONS_FIELD_NAME|documents|Default field name to process the citations output from Promptflow request.|
 |DATASOURCE_TYPE||Type of data source to use for using the 'on-your-data' api. Can be `AzureCognitiveSearch`, `AzureCosmosDB`, `Elasticsearch`, `Pinecone`, `AzureMLIndex`, `AzureSqlServer` or `None` |
-
+|AZURE_SQL_SERVER_PORT||Not publicaly available at this time.|
+|AZURE_SQL_SERVER_DATABASE_NAME||Not publicaly available at this time.|
```
Comment (sarah-widder) [inline README.md:269]: typo - publicly

File: backend/settings.py [hunk 2]
```diff
@@ -123,6 +123,7 @@ class _AzureOpenAISettings(BaseSettings):
     embedding_endpoint: Optional[str] = None
     embedding_key: Optional[str] = None
     embedding_name: Optional[str] = None
+    chatgpt_key: Optional[str] = None
```
Comment (sarah-widder) [inline backend/settings.py:126]: remove this one

**Review by abhahn — DISMISSED** [review]
Overall looks great!  Just a couple of small things I have left comments on.  I also am having some issues starting up, but it may be a config / RBAC issue for me, so I'm going to leave this review as-is for now and investigate separately on my side to see if I can get things working.

File: README.md [hunk 3]
```diff
@@ -391,6 +391,60 @@ We recommend keeping these best practices in mind:
...
+|PROMPTFLOW_ENDPOINT||URL of the deployed Promptflow endpoint e.g. https://pf-deployment-name.region.inference.ml.azure.com/score|
+|PROMPTFLOW_API_KEY||Auth key for deployed Promptflow endpoint. Note: only Key-based authentication is supported.|
+|PROMPTFLOW_RESPONSE_TIMEOUT|120|Timeout value in seconds for the Promptflow endpoint to respond.|
+|PROMPTFLOW_REQUEST_FIELD_NAME|query|Default field name to construct Promptflow request. Note: chat_history is auto constucted based on the interaction, if your API expects other mandatory field you will need to change the request parameters under `promptflow_request` function.|
+|PROMPTFLOW_RESPONSE_FIELD_NAME|reply|Default field name to process the response from Promptflow request.|
+|PROMPTFLOW_CITATIONS_FIELD_NAME|documents|Default field name to process the citations output from Promptflow request.|
+|DATASOURCE_TYPE||Type of data source to use for using the 'on-your-data' api. Can be `AzureCognitiveSearch`, `AzureCosmosDB`, `Elasticsearch`, `Pinecone`, `AzureMLIndex`, `AzureSqlServer` or `None` |
+|AZURE_SQL_SERVER_PORT||Not publicly available at this time.|
```
[the remark this answers is not in the document]
Comment (iseabock) [reply README.md:445]: My understanding is that it is not available at all publicly and for internal testing right now.

File: README.md [hunk 4]
```diff
@@ -391,6 +391,60 @@ We recommend keeping these best practices in mind:
 
 **A note on Azure OpenAI API versions**: The application code in this repo will implement the request and response contracts for the most recent preview API version supported for Azure OpenAI.  To keep your application up-to-date as the Azure OpenAI API evolves with time, be sure to merge the latest API version update into your own application code and redeploy using the methods described in this document.
 
+## Environment variables
```
[the remark this answers is not in the document]
Comment (sarah-widder) [reply README.md:423]: I think remove this section from the PR, since the variables are already discussed elsewhere

**Review by sarah-widder — APPROVED** [review]

**Pull request merged**